File: frontend/src/microservices/operations/airline/objects/ManagerFlightOrderTravelerResponse.ts
// 鏈枃浠跺畾涔?operations 妯″潡鐨?`ManagerFlightOrderTravelerResponse`锛屼綔涓哄搷搴旀暟鎹苟鎻愪緵 JSON 缂栬В鐮併€?

import type {
  TravelerBasicInfo,
  TravelerContactInfo,
  TravelerDocumentInfo,
  TravelerPreferenceInfo,
  TravelerServiceSummary,
  TravelerSpecialRequirementInfo,
} from '@/microservices/traveler/objects/TravelerProfileDetails'

export type ManagerFlightOrderTravelerResponse = {
  travelerId: string
  fullName: string
  documentNumber: string
  basicInfo: TravelerBasicInfo
  documentInfo: TravelerDocumentInfo
  contactInfo: TravelerContactInfo
  preferenceInfo: TravelerPreferenceInfo
  specialRequirementInfo: TravelerSpecialRequirementInfo
  serviceSummary: TravelerServiceSummary
}

export const managerFlightOrderTravelerResponseFromJson = (json: string): ManagerFlightOrderTravelerResponse =>
  JSON.parse(json) as ManagerFlightOrderTravelerResponse

export const managerFlightOrderTravelerResponseToJson = (value: ManagerFlightOrderTravelerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/ManagerFlightOrderTravelerResponse.scala
// 本文件定义 airline 订单中旅客明细的响应结构，用于航司管理端查看订单。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerFlightOrderTravelerResponse(
    travelerId: String,
    fullName: String,
    documentNumber: String,
    basicInfo: ManagerTravelerBasicInfo,
    documentInfo: ManagerTravelerDocumentInfo,
    contactInfo: ManagerTravelerContactInfo,
    preferenceInfo: ManagerTravelerPreferenceInfo,
    specialRequirementInfo: ManagerTravelerSpecialRequirementInfo,
    serviceSummary: ManagerTravelerServiceSummary
)
object ManagerFlightOrderTravelerResponse:
  given sourceEncoder: Encoder[ManagerFlightOrderTravelerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerFlightOrderTravelerResponse] = deriveDecoder
